### PR TITLE
Save for later: toggle button in reader tiles

### PR DESCRIPTION
**Pull request by ctarda**
Implements #9247 

**This PR does not have a milestone yet**. I definitely don't want to get it into the rush for the 10.0 release.

This is an early PR to gather feedback on the general direction it is taking. 

simulator screen shot - iphone 8 - 2018-05-07 at 12 50 24

This PR adds a toggle button to the tiles in the reader stream, in the place occupied until now for the Share button. As part of the PR, I have also added some stubbed methods in the `ReaderPostService` and the `ReaderPost` object. The implementation if the service will have to be attacked [here](https://github.com/wordpress-mobile/WordPress-iOS/issues/9265), and the implementation of the model object [here](https://github.com/wordpress-mobile/WordPress-iOS/issues/9227).

After some discussion about the way we want the data model to be, in this PR we diverge from the original plan by embracing a boolean flag in `ReaderPost` instead of adding a separate model object to model posts saved for later.

To test:
1. Deactivate the feature flag
2. Navigate to reader. The Share button should be visible
3. Activate the feature flag
4. Navigate to reader. The bookmark button should be visible instead of the Share button
5. Tapping the bookmark button should make it change color (to blue). The `ReaderPostService` should log a message in the console (to be removed when it is actually implemented)

### CODE REVIEW

File: WordPress/Classes/Services/ReaderPostService.m [hunk 1]
```diff
@@ -371,6 +371,15 @@ - (void)toggleFollowingForPost:(ReaderPost *)post success:(void (^)(void))succes
     }
 }
 
+- (void)toggleSavedForLaterForPost:(ReaderPost *)post
+                           success:(void (^)(void))success
+                           failure:(void (^)(NSError *error))failure
+{
+    NSLog(@"==== Toggling save for later for post %", post);
```
Comment (frosty) [inline WordPress/Classes/Services/ReaderPostService.m]: Nitpick, but I'd remove this comment, or use `DDLogDebug`.

File: WordPress/Classes/ViewRelated/Reader/ReaderPostCardCell.swift [hunk 2]
```diff
@@ -408,11 +409,13 @@ fileprivate func < <T: Comparable>(lhs: T?, rhs: T?) -> Bool {
         if contentProvider == nil || contentProvider?.sourceAttributionStyle() != SourceAttributionStyle.none {
             resetActionButton(commentActionButton)
             resetActionButton(likeActionButton)
+            resetActionButton(saveForLaterButton)
             return
         }
 
         configureCommentActionButton()
         configureLikeActionButton()
+        configureSaveForLaterButton()
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Reader/ReaderPostCardCell.swift]: We should only call this (or add a guard into the method) if the feature flag is enabled. I know currently the button won't be highlighted, as `isSavedForLater` will be false, but we may end up putting more things in that method later.

File: WordPress/Classes/ViewRelated/Reader/ReaderPostCardCell.swift [hunk 3]
```diff
@@ -660,9 +673,9 @@ extension ReaderPostCardCell: Accessible {
     }
 
     private func prepareShareForVoiceOver() {
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Reader/ReaderPostCardCell.swift]: We should update these according to the feature flag enabled state for saveForLater.

File: WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift [hunk 4]
```diff
@@ -839,6 +839,12 @@ import WordPressFlux
                                         })
     }
 
+    fileprivate func toggleSavedForLater(for post: ReaderPost) {
+        SaveForLaterAction().execute(with: post, context: managedObjectContext(), completion: { [weak self] in
+            self?.syncHelper.syncContent()
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift]: Do we need to sync content currently if save for later is local only?

File: WordPress/Classes/ViewRelated/Reader/SaveForLaterAction.swift [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        present(notice)
+    }
+
+    private func presentPostRemovedNotice(for post: ReaderPost, context: NSManagedObjectContext, completion: @escaping () -> Void) {
+        /// I am not sure if this could just be solved by invoking undo in the context's undo manager. Does ReaderStreamViewController observe the context in any way? In the meantime, here is a very imperative implementation of the undo
+        let notice = Notice(title: Strings.postRemoved,
+                            feedbackType: .success,
+                            actionTitle: Strings.undo,
+                            actionHandler: {
+                                self.toggleSavedForLater(post, context: context, completion: completion)
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Reader/SaveForLaterAction.swift]: I think this is fine.

File: WordPress/Classes/ViewRelated/Reader/SaveForLaterAction.swift [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+        present(notice)
+    }
+
+    private func present(_ notice: Notice) {
+        ActionDispatcher.dispatch(NoticeAction.post(notice))
+    }
+
+    private func showAll() {
+        //Navigate to all saved for later
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Reader/SaveForLaterAction.swift]: Could you change this to a TODO: to help us find it later?

File: WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift (hunk 4 above)
Comment (ctarda) [reply WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift]: I have no idea, to be honest, but I'll remove it for now. What does `syncContent` do exactly?

**Comment by ctarda** [comment]
And the button should toggle its state now

**Pull request merged**